### DISCUSSION TITLE
Strange graph rendering

### DISCUSSION THREAD

**Discussion by xcelm00** [Megabit/Blazorise#2344] [Q&A]
Hi, I am using net Core with blazor and Blazorise for charts (they look awesome btw :-) ). 

I stumbled upon 2 problems with rendering the charts. I looked in the documentation and I added method 
protected override async Task OnAfterRenderAsync(bool firstRender) (see in the code below) 

The problem is that this method always fires twice. It renders the charts in the first go and the second time it leaves them empty (as firstRender = false the second time it fires). If I remove the if block the charts render ok.

Furthermore I've added button that should refresh the data and charts. After pressing this button the charts refresh twice (this is unwanted behaviour as it distracts the users) and what is interesting the data itself (the values) change after the second go.

Have anybody dealt with this problem before?

My html button to refresh the data:

`<div class="btn" @onclick="(async () => await RerenderPage())">Refresh Data</div>
`

My c# code:

```
List<Models.VM.OverView> overview = new List<Models.VM.OverView>();

protected override async Task OnInitializedAsync()
{
    overview = await GetOverview(); //gets overview from api
}
        
public async Task<List<Models.VM.OverView>> GetOverview()
{
 return await Http.GetFromJsonAsync<List<Models.VM.OverView>>("/api/Overview/GetOverView/" + await GetUserIdAsync);
}
        
protected override async Task OnAfterRenderAsync(bool firstRender)
{
     if (firstRender)
     {
         await HandleRedraw();
     }        
}
        
async Task HandleRedraw()
{
    await pieChart.Clear();


    //this method goes in the overview object and gets data from it
    await pieChart.AddLabelsDatasetsAndUpdate(GetLabelsPieChart(), GetPieChartDataset());
}
```

**Comment by stsrki** [comment]
I believe I had the same problem when I was creating a demo for the charts. So I just used a member variable instead of using the `firstRender` flag.

Look here for reference https://github.com/stsrki/Blazorise/blob/master/Demos/Blazorise.Demo/Pages/Tests/ChartsPage.razor#L113

**Comment by wartim** [comment]
I'd like to know why the "first render" judgment fails
